src/main/java/wooteco/subway/service/StationService.java

```diff
@@ -0,0 +1,49 @@
+package wooteco.subway.service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import wooteco.subway.dao.StationDao;
+import wooteco.subway.domain.Station;
+import wooteco.subway.dto.StationRequest;
+import wooteco.subway.dto.StationResponse;
+
+@Service
+public class StationService {
+
+    static final String DUPLICATE_EXCEPTION_MESSAGE = "이름이 중복된 역은 만들 수 없습니다.";
+
+    private final StationDao stationDao;
+
+    public StationService(StationDao stationDao) {
+        this.stationDao = stationDao;
+    }
+
+    @Transactional
+    public StationResponse insertStation(StationRequest stationRequest) {
+        Station station = stationRequest.toEntity();
+        validateDuplicateName(station);
+        Station newStation = stationDao.insert(station);
+        return new StationResponse(newStation);
+    }
+
+    private void validateDuplicateName(Station station) {
+        if(stationDao.existByName(station)){
+            throw new IllegalArgumentException(DUPLICATE_EXCEPTION_MESSAGE);
+        }
+    }
+
+    public List<StationResponse> findStations() {
+        List<Station> stations = stationDao.findAll();
+        List<StationResponse> stationResponses = stations.stream()
+                .map(StationResponse::new)
+                .collect(Collectors.toUnmodifiableList());
+        return stationResponses;
```

Comment:
```suggestion
        return stations.stream()
                .map(StationResponse::new)
                .collect(Collectors.toUnmodifiableList());
```

위와 같이 인라인으로 쓰는게 더 깔끔한 경우가 있더라구요 :) 이건 레넌이 한번 판단해보세요~

Reply:
스트림처럼 데이터를 가공하는 과정은 복잡하면 따로 상수로 빼서 이름으로 의미를 나타내주는 것이 좋다고 생각합니다. 해당 스트림은 복잡하지 않으니 럿고 말씀대로 인라인으로 써도 괜찮을 것 같아요!